lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx

```diff
@@ -1,24 +1,61 @@
 import React from 'react';
-import {isFromContact, Message} from '../../../httpclient/model';
+import {Message} from '../../../httpclient/model';
 import {getDefaultMessageRenderingProps, MessageRenderProps} from '../../shared';
+import {RichCard} from '../../components/RichCard';
+import {RichCardContent, ContentUnion, TextContent} from './chatPluginModel';
 import {Text} from '../../components/Text';
-import {ContentUnion} from './chatPluginModel';
 
 export const ChatPluginRender = (props: MessageRenderProps) => {
-  const message = props.message;
-  return render(mapContent(message), props);
+  const {
+    message: {content},
+    message,
+  } = props;
+
+  if (JSON.parse(content).text) {
+    return render(mapTextContent(message), props);
+  }
+
+  if (JSON.parse(content).richCard) {
+    return render(mapRichCardContent(message), props);
+  }
 };
 
 function render(content: ContentUnion, props: MessageRenderProps) {
   switch (content.type) {
     case 'text':
       return <Text {...getDefaultMessageRenderingProps(props)} text={content.text} />;
+    case 'richCard':
+      return (
+        <RichCard
+          {...getDefaultMessageRenderingProps(props)}
+          title={content.title}
+          description={content.description}
+          media={content.media}
+          suggestions={content.suggestions}
+        />
+      );
   }
 }
 
-function mapContent(message: Message): ContentUnion {
+function mapTextContent(message: Message): TextContent {
   return {
     type: 'text',
     text: JSON.parse(message.content).text,
   };
 }
+
+function mapRichCardContent(message: Message): RichCardContent {
```

Comment:
Ah, so I think that there was a slight misunderstanding about the meaning of "mapping" :) The idea is that there is only one function that 1. figures out the content type and 2) returns the correct `ContentUnion` subtype (while the method as a whole returns `ContentUnion`). That's why I called the method generically `mapContent`.

With this approach, you end up writing the complicated code of figuring out what type of content a message is in the `ChatPluginRender` and you also call `JSON.parse` multiple times.

Reply:
yeah I wanted to have specific functions to "map" each type, but you're right, it ends up repeating code and complicated things. Fixing it right now!